core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -301,15 +301,24 @@ public function from($identifier): self
             return $this->fromImage($identifier);
         }
 
-        if ($this->validatorAdapter()->isUuid($identifier)) {
+        $isString = \is_string($identifier);
+
+        if ($isString && $this->validatorAdapter()->isUuid($identifier)) {
             return $this->fromUuid($identifier);
         }
 
         if (is_numeric($identifier)) {
             return $this->fromId((int) $identifier);
         }
 
-        return $this->fromPath($identifier);
+        if ($isString) {
+            return $this->fromPath($identifier);
+        }
+
+        $method = __METHOD__;
+        $type = \is_object($identifier) ? \get_class($identifier) : \gettype($identifier);
+
+        throw new \TypeError("$method(): Argument #1 (\$identifier) must be of type FilesModel|ImageInterface|string|int|null, $type given");
```

Comment:
```suggestion
        throw new \TypeError(sprintf(
            '%s(): Argument #1 ($identifier) must be of type FilesModel|ImageInterface|string|int|null, "%s" given',
            ___METHOD__,
            \is_object($identifier) ? \get_class($identifier) : \gettype($identifier)
        ));
```

That is the classic `sprintf()` case IMHO. 😄

Reply:
That's right, but then the CS fixer tries to cram everything in a single line. :smile: 
I'll change it.

Reply:
Ah, I see. Then we might want to keep `$type = \is_object($identifier) ? \get_class($identifier) : \gettype($identifier)` separately?

Reply:
See 55cd2839d, IMO that's a bit neater.